javaslang/src/main/java/javaslang/collection/IndexedSeq.java

```diff
@@ -221,16 +223,68 @@ default int lastIndexWhere(Predicate<? super T> predicate, int end) {
     @Override
     default Iterator<T> reverseIterator() {
         return new AbstractIterator<T>() {
-            private int i = IndexedSeq.this.length();
+            final ListIterator<T> delegate = IndexedSeq.this.listIterator(IndexedSeq.this.length());
 
             @Override
             public boolean hasNext() {
-                return i > 0;
+                return delegate.hasPrevious();
             }
 
             @Override
             public T getNext() {
-                return IndexedSeq.this.get(--i);
+                return delegate.previous();
+            }
+        };
+    }
+
+    default ListIterator<T> listIterator(int index) {
```

Comment:
We should not add additional API to our collections without having it discussed first! Especially we will not add a method that returns a java.util.ListIterator. Javaslang's collections are designed to co-exist beside Java's standard-collections. There are well-defined conversion methods between them but adding operations with types 'from the other side' mixes it up too much. ==> ListIterator is meant to be used with a java.util.List.